src/field/pokemon.ts

```diff
@@ -3053,12 +3055,13 @@ export class PokemonSummonData {
 export class PokemonBattleData {
   public hitCount: integer = 0;
   public endured: boolean = false;
-  public berriesEaten: BerryType[] = [];
+  public berriesEaten: BerryType[] = []; 
 }
 
 export class PokemonBattleSummonData {
   public turnCount: integer = 1;
   public moveHistory: TurnMove[] = [];
+  public abilityTriggeredThisSwitch: boolean = false;
```

Comment:
```suggestion
  public abilityTriggered: boolean = false;
```
Being in battle summon data, I believe 'this switch' is implied